src/umpire/alloc/PosixMemalignAllocator.hpp

```diff
@@ -0,0 +1,72 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_PosixMemalignAllocator_HPP
+#define UMPIRE_PosixMemalignAllocator_HPP
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "umpire/util/Macros.hpp"
+
+namespace umpire {
+namespace alloc {
+
+/*!
+ * \brief Uses posix_memalign() and free() to allocate page-aligned memory.
+ */
+struct PosixMemalignAllocator
+{
+  /*!
+   * \brief Allocate bytes of memory using posix_memalign.
+   *
+   * \param bytes Number of bytes to allocate. Does not have to be a multiple of the system page size.
+   * \return Pointer to start of the allocation.
+   *
+   * \throws umpire::util::Exception if memory cannot be allocated.
+   */
+  void* allocate(size_t bytes)
+  {
+    const long pagesize = sysconf(_SC_PAGESIZE);
```

Comment:
Can we cache this? I don't know if this call is expensive

Reply:
Sure. I didn't do that initially because there's no clear location for it, since this is in a header file. I'll add some NUMA wrapper methods in `util/Numa.{c,h}pp` and drop it in there 👍

Reply:
You could just make it a `static`, but either way is fine.